clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2369,8 +2369,8 @@ protected void processCommonConsumerAction(ConsumerAction consumerAction) throws
          * {@link #kafkaDataValidationService}, we would like to drain all the buffered messages before cleaning up those
          * two variables to avoid the race condition.
          */
-        partitionConsumptionStateMap.remove(partition);
         if (consumerAction.isHelixTriggeredAction()) {
+          partitionConsumptionStateMap.remove(partition);
```

Comment:
If it is not removed, would that be cleaned up and reinitialized during error reset? 
Also for unsubscribe action it is going to stop ingestion but it is still taking up disk space?

Reply:
IIUC, a reset should trigger an O→S state transition, which corresponds to a SUBSCRIBE action. SUBSCRIBE flow should create a new PCS and replace the old one.